network/network.go

```diff
@@ -557,8 +525,7 @@ func (n *network) Dispatch() error {
 	go n.inboundConnUpgradeThrottler.Dispatch()
 	defer n.inboundConnUpgradeThrottler.Stop()
 	go func() {
-		compatibility := n.compatibility()
```

Comment:
I didn't quite understand the places where `compatibility()` is replaced by `versionCompatibility`. Could you elaborate on why it was done?

Reply:
This was a change I originally did myself, to have support for old nodes, which were still sending legacy Avalanche version information, and new nodes, which were sending the correct Flare version information. Now that all nodes are on the new code, we can drop support for the legacy version.

`compatibility()` picked one compatibility version based on what hard fork we were at.

`compatibilities()` picked one or two compatibility versions based on what hard fork we were at, and had accompanying functions to mirror the compatibility functions for more than one compatibility version.